File: src/main/java/com/github/nsc/de/shake/parser/node/expression/PriorityNode.java
package com.github.nsc.de.shake.parser.node.expression;

import com.github.nsc.de.shake.parser.node.ValuedNode;

public class PriorityNode implements ValuedNode {

    private final ValuedNode value;

    public PriorityNode(ValuedNode value) {
        this.value = value;
    }

    public ValuedNode getValue() {
        return value;
    }

    @Override
    public String toString() {
        return "PriorityNode{" +
                "value=" + value +
                '}';
    }
}

File: parser/src/commonMain/kotlin/io/github/shakelang/shake/parser/node/factor/PriorityNode.kt
package io.github.shakelang.shake.parser.node.factor

import io.github.shakelang.shake.parser.node.ValuedNode
import io.github.shakelang.parseutils.characters.position.PositionMap

class PriorityNode(map: PositionMap, val value: ValuedNode) : ValuedNode(map) {
    override fun toJson(): Map<String, *> = mapOf("name" to "PriorityNode", "value" to value)
}